src/vivarium_gates_child_iv_iron/data/utilities.py

```diff
@@ -0,0 +1,341 @@
+    data = _scrub_age(data, age_group_ids)
+    data = vi_utils.scrub_year(data)
+    data = vi_utils.scrub_affected_entity(data)
+    return data
+
+
+def _scrub_age(data: pd.DataFrame, age_group_ids: List[int] = None) -> pd.DataFrame:
+    if 'age_group_id' in data.index.names:
+        age_bins = get_gbd_age_bins(age_group_ids).set_index('age_group_id')
+        id_levels = data.index.levels[data.index.names.index('age_group_id')]
+        interval_levels = [pd.Interval(age_bins.age_start[age_id], age_bins.age_end[age_id], closed='left')
+                           for age_id in id_levels]
+        data.index = data.index.rename('age', 'age_group_id').set_levels(interval_levels, 'age')
+    return data
+
+
+def get_gbd_age_bins(age_group_ids: List[int] = None) -> pd.DataFrame:
+    # If no age group ids are specified, use the standard GBD 2019 age bins
+    if not age_group_ids:
+        age_group_ids = gbd.get_age_group_id()
+    # from gbd.get_age_bins()
+    q = f"""
+                SELECT age_group_id,
+                       age_group_years_start,
+                       age_group_years_end,
+                       age_group_name
+                FROM age_group
+                WHERE age_group_id IN ({','.join([str(a) for a in age_group_ids])})
+                """
+    raw_age_bins = query(q, 'shared')
+
+    # from utility_data.get_age_bins()
+    age_bins = (
+        raw_age_bins[['age_group_id', 'age_group_name', 'age_group_years_start', 'age_group_years_end']]
+        .rename(columns={'age_group_years_start': 'age_start', 'age_group_years_end': 'age_end'})
+    )
+
+    # set age start for birth prevalence age bin to -1 to avoid validation issues
+    age_bins.loc[age_bins['age_end'] == 0.0, 'age_start'] = -1.0
+    return age_bins
+
+
+def filter_relative_risk_to_cause_restrictions(data: pd.DataFrame) -> pd.DataFrame:
+    """ It applies age restrictions according to affected causes
+    and affected measures. If affected measure is incidence_rate,
+    it applies the yld_age_restrictions. If affected measure is
+    excess_mortality_rate, it applies the yll_age_restrictions to filter
+    the relative_risk data"""
+
+    temp = []
+    affected_entities = set(data.affected_entity)
+    affected_measures = set(data.affected_measure)
+    for cause, measure in product(affected_entities, affected_measures):
+        df = data[(data.affected_entity == cause) & (data.affected_measure == measure)]
+        cause = get_entity(EntityKey(f'cause.{cause}.{measure}'))
+        if measure == 'excess_mortality_rate':
+            start, end = vi_utils.get_age_group_ids_by_restriction(cause, 'yll')
+        else:  # incidence_rate
+            start, end = vi_utils.get_age_group_ids_by_restriction(cause, 'yld')
+        temp.append(df[df.age_group_id.isin(range(start, end + 1))])
+    data = pd.concat(temp)
+    return data
+
+
+def get_intervals_from_categories(lbwsg_type: str, categories: Dict[str, str]) -> pd.Series:
```

Comment:
Separated this function out from LBWSG abstract base class.  Using string for the "sub risk" to determine which parsing function to use for given sub risk.